test/permissions/remote.provider.getDetails.js

```diff
@@ -0,0 +1,213 @@
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for category in Many decorator',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'Permission',
+          permissions.S1.ReadOnlyCategory,
+          'Actions',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission for category with subsystem/catalog, sufficient permission',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.AdminUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          users.S1.AdminUser,
+          'Roles',
+          isSuccess(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission for category with subsystem/catalog, no record in category',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.AdminUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          roles.S1.Admin,
+          'Roles',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for category with subsystem/catalog',
+    (test, { remoteProvider, login }) =>
+      login(users.S2.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          users.S2.ReadOnlyUser,
+          'Roles',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for subsystem/catalog',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          users.S2.ReadOnlyUser,
+          'Roles',
+          isError(test)
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for category with subsystem/catalog in Many decorator',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.ReadOnlyUser, () =>
+        remoteProvider.getDetails(
+          'SystemUser',
+          users.S1.ReadOnlyUser,
+          'Roles',
+          (err, res) => {
+            test.error(err);
+            test.assertNot(res[0]);
+            test.end();
+          }
+        )
+      )
+  );
+
+  test.test(
+    'remoteProvider.getDetails: check read permission, no read permission for subsystem/catalog in Many decorator',
+    (test, { remoteProvider, login }) =>
+      login(users.S1.AdminUser, () =>
+        provider.linkDetails(
+          'SystemUser',
+          'Roles',
+          users.S1.GuestUser,
+          roles.S2.Guest,
+          test.cb(() =>
+            remoteProvider.getDetails(
+              'SystemUser',
+              users.S1.GuestUser,
+              'Roles',
+              (err, res) => {
+                test.error(err);
```

Comment:
Ditto `test.cb` then =).

Reply:
@lundibundi haven't done this here because of linter `handle-callback-err` rule, as callback in `test.cb` have to take `err` as well, but I don't need it . Should I disable here this rule and use `test.cb`?

Reply:
You can name the argument anything else but `err`, `error` etc and it won't complain, like `(_, res)` or use `cbFail`. Or you can leave it as is, up to you.